src/sql/base/browser/ui/modal/modal.ts

```diff
@@ -305,17 +305,21 @@ export abstract class Modal extends Disposable implements IThemable {
 			}
 		});
 		this._resizeListener = DOM.addDisposableListener(window, DOM.EventType.RESIZE, (e: Event) => {
-			this.layout(DOM.getTotalHeight(this._builder.getHTMLElement()));
+			this._layout();
 		});
 
-		this.layout(DOM.getTotalHeight(this._builder.getHTMLElement()));
+		this._layout();
 		TelemetryUtils.addTelemetry(this._telemetryService, TelemetryKeys.ModalDialogOpened, { name: this._name });
 	}
 
 	/**
 	 * Required to be implemented so that scrolling and other functions operate correctly. Should re-layout controls in the modal
 	 */
-	protected abstract layout(height?: number): void;
+	protected abstract layout(dimension: Dimension): void;
+
+	protected _layout(): void {
```

Comment:
>_layout [](start = 11, length = 7)

Should rename this function. _layout and layout is confusing especially calling from another class. Maybe LayoutWithBodyContent?

Reply:
the layout(dimension) method falls in line with vscode's ui elements of having a layout method with a measurement. the _layout was basically just a helper function that grabs all the measurements for the modal and calls the correct layout method